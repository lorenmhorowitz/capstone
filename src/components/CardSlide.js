import React, { Component } from "react";
import "../css/CardSlide.css";
import { Carousel } from "react-responsive-carousel";

class CardSlide extends Component {
  constructor(props) {
    super(props);

    this.state = {
      slides: []
    };

    this.initializePictures();
  }

  initializePictures = () => {
    const pictures = [];

    let x = 1;
    this.props.images.forEach(image => {
      const picture = (
        <div key={x}>
          <img
            style={{ width: this.props.width, height: this.props.height }}
            src={image}
          />
        </div>
      );
      pictures.push(picture);
      x += 1;
    });
    this.setState({
      slides: pictures
    });
  };

  render() {
    return (
      <div>
        <Carousel infiniteLoop={true} showThumbs={false} dynamicHeight={true}>
          {this.state.slides}
        </Carousel>
      </div>
    );
  }
}

export default CardSlide;
