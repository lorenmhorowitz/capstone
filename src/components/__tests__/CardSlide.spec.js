import React from "react";
import CardSlide from "../CardSlide";
import { shallow } from "enzyme";

describe("CardSlide test component", () => {
  it("renders correctly", () => {
    const images = [
      "https://images.pexels.com/photos/106399/pexels-photo-106399.jpeg?auto=compress&cs=tinysrgb&h=350",
      "https://www.control4.com/files/large/e949244caec63b0c7d8c5edd86b8ee95.jpeg"
    ];
    const wrapper = shallow(
      <CardSlide images={images} height={250} width={250} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
