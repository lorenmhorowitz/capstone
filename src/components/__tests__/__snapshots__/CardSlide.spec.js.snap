// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSlide test component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardSlide
    height={250}
    images={
      Array [
        "https://images.pexels.com/photos/106399/pexels-photo-106399.jpeg?auto=compress&cs=tinysrgb&h=350",
        "https://www.control4.com/files/large/e949244caec63b0c7d8c5edd86b8ee95.jpeg",
      ]
    }
    width={250}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Carousel
        autoPlay={false}
        axis="horizontal"
        centerMode={false}
        centerSlidePercentage={80}
        dynamicHeight={true}
        emulateTouch={false}
        infiniteLoop={true}
        interval={3000}
        onChange={[Function]}
        onClickItem={[Function]}
        onClickThumb={[Function]}
        selectedItem={0}
        showArrows={true}
        showIndicators={true}
        showStatus={true}
        showThumbs={false}
        statusFormatter={[Function]}
        stopOnHover={true}
        swipeScrollTolerance={5}
        swipeable={true}
        transitionTime={350}
        useKeyboardArrows={false}
        verticalSwipe="standard"
        width="100%"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoPlay": false,
        "axis": "horizontal",
        "centerMode": false,
        "centerSlidePercentage": 80,
        "children": Array [],
        "dynamicHeight": true,
        "emulateTouch": false,
        "infiniteLoop": true,
        "interval": 3000,
        "onChange": [Function],
        "onClickItem": [Function],
        "onClickThumb": [Function],
        "selectedItem": 0,
        "showArrows": true,
        "showIndicators": true,
        "showStatus": true,
        "showThumbs": false,
        "statusFormatter": [Function],
        "stopOnHover": true,
        "swipeScrollTolerance": 5,
        "swipeable": true,
        "transitionTime": 350,
        "useKeyboardArrows": false,
        "verticalSwipe": "standard",
        "width": "100%",
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Carousel
          autoPlay={false}
          axis="horizontal"
          centerMode={false}
          centerSlidePercentage={80}
          dynamicHeight={true}
          emulateTouch={false}
          infiniteLoop={true}
          interval={3000}
          onChange={[Function]}
          onClickItem={[Function]}
          onClickThumb={[Function]}
          selectedItem={0}
          showArrows={true}
          showIndicators={true}
          showStatus={true}
          showThumbs={false}
          statusFormatter={[Function]}
          stopOnHover={true}
          swipeScrollTolerance={5}
          swipeable={true}
          transitionTime={350}
          useKeyboardArrows={false}
          verticalSwipe="standard"
          width="100%"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoPlay": false,
          "axis": "horizontal",
          "centerMode": false,
          "centerSlidePercentage": 80,
          "children": Array [],
          "dynamicHeight": true,
          "emulateTouch": false,
          "infiniteLoop": true,
          "interval": 3000,
          "onChange": [Function],
          "onClickItem": [Function],
          "onClickThumb": [Function],
          "selectedItem": 0,
          "showArrows": true,
          "showIndicators": true,
          "showStatus": true,
          "showThumbs": false,
          "statusFormatter": [Function],
          "stopOnHover": true,
          "swipeScrollTolerance": 5,
          "swipeable": true,
          "transitionTime": 350,
          "useKeyboardArrows": false,
          "verticalSwipe": "standard",
          "width": "100%",
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
